integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py

```diff
@@ -95,4 +104,80 @@ def __init__(
             api_base_url=api_base_url,
             organization=None,
             generation_kwargs=generation_kwargs,
+            tools=tools,
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
could you explain why this method and the following one need to be overridden?

Reply:
yeah, of course. We have a mismatch with super class when it comes to init params. So it seems like we have two options:
1) Match the params from OpenAIChatGenerator
2) Adjust serialization of only selected parameters

In case of 2, if we don't do this, the serde tests were failing because we were init-ing params like max_retries from super class. 
We can also do option 1) and remove to_dict()

wdyt?

Reply:
ok I think that your strategy makes sense. Please add a short comment in the code explaining this, so we remember in future.